src/daemon/Display.cpp

```diff
@@ -121,9 +121,21 @@ namespace SDDM {
             // set flags
             m_started = true;
 
+            // determine session type
+            const QString &autologinSession = mainConfig.Autologin.Session.get();
+            Session session;
+            if (findSessionEntry(mainConfig.XDisplay.SessionDir.get(), autologinSession)) {
+                session.setTo(Session::X11Session, autologinSession);
+            } else if (findSessionEntry(mainConfig.WaylandDisplay.SessionDir.get(), autologinSession)) {
+                session.setTo(Session::WaylandSession, autologinSession);
+            } else {
+                qCritical() << "Unable to find autologin session entry" << autologinSession;
+                return;
```

Comment:
you don't want to return here, we need login to attempt and fail so that we hanlde it and show the greeter the second time.

This version will just hang (I think)